src/rust/engine/src/externs/interface.rs

```diff
@@ -832,89 +828,89 @@ async fn workunit_to_py_value(
+        UserMetadataItem::PyValue(py_val_handle) => (**py_val_handle)
+          .as_any()
+          .downcast_ref::<Value>()
+          .ok_or_else(|| {
+            PyException::new_err(format!("Failed to convert {py_val_handle:?} to a Value."))
+          })?
+          .to_object(py),
+      };
+      user_metadata_entries.push((
+        externs::store_utf8(py, user_metadata_key.as_str()),
+        Value::new(value),
+      ));
+    }
+
+    dict_entries.push((
+      externs::store_utf8(py, "metadata"),
+      externs::store_dict(py, user_metadata_entries)?,
     ));
-  }
 
-  dict_entries.push((
-    externs::store_utf8(py, "metadata"),
-    externs::store_dict(py, user_metadata_entries)?,
-  ));
+    if let Some(stdout_digest) = metadata.stdout {
+      artifact_entries.push((
+        externs::store_utf8(py, "stdout_digest"),
+        crate::nodes::Snapshot::store_file_digest(py, stdout_digest)
+          .map_err(PyException::new_err)?,
+      ));
+    }
 
-  if let Some(stdout_digest) = metadata.stdout {
-    artifact_entries.push((
-      externs::store_utf8(py, "stdout_digest"),
-      crate::nodes::Snapshot::store_file_digest(py, stdout_digest).map_err(PyException::new_err)?,
-    ));
-  }
+    if let Some(stderr_digest) = metadata.stderr {
+      artifact_entries.push((
+        externs::store_utf8(py, "stderr_digest"),
+        crate::nodes::Snapshot::store_file_digest(py, stderr_digest)
+          .map_err(PyException::new_err)?,
+      ));
+    }
 
-  if let Some(stderr_digest) = metadata.stderr {
-    artifact_entries.push((
-      externs::store_utf8(py, "stderr_digest"),
-      crate::nodes::Snapshot::store_file_digest(py, stderr_digest).map_err(PyException::new_err)?,
+    dict_entries.push((
+      externs::store_utf8(py, "artifacts"),
+      externs::store_dict(py, artifact_entries)?,
     ));
-  }
-
-  dict_entries.push((
-    externs::store_utf8(py, "artifacts"),
-    externs::store_dict(py, artifact_entries)?,
-  ));
 
-  // TODO: Temporarily attaching the global counters to the "root" workunit. Callers should
-  // switch to consuming `StreamingWorkunitContext.get_metrics`.
-  // Remove this deprecation after 2.14.0.dev0.
-  if !has_parent_ids {
-    let mut metrics = workunit_store.get_metrics();
+    // TODO: Temporarily attaching the global counters to the "root" workunit. Callers should
+    // switch to consuming `StreamingWorkunitContext.get_metrics`.
+    // Remove this deprecation after 2.14.0.dev0.
+    if !has_parent_ids {
+      let mut metrics = workunit_store.get_metrics();
 
-    metrics.insert("DEPRECATED_ConsumeGlobalCountersInstead", 0);
-    let counters_entries = metrics
-      .into_iter()
-      .map(|(counter_name, counter_value)| {
-        (
-          externs::store_utf8(py, counter_name),
-          externs::store_u64(py, counter_value),
-        )
-      })
-      .collect();
+      metrics.insert("DEPRECATED_ConsumeGlobalCountersInstead", 0);
```

Comment:
This just seems like whitespace changes?

Reply:
The diff is confusing. Did these move within a newly-introduced block?

Reply:
Yup. A `Python::with_python` block

Reply:
Line 843 above